lib/render/SliceRenderer.cpp

```diff
@@ -140,6 +136,16 @@ void SliceRenderer::_resetCache()
 
     _getExtents(_cacheParams.boxMin, _cacheParams.boxMax);
 
+    // clang-format off
+    _dataMgr->GetVariableExtents(_cacheParams.ts, 
```

Comment:
It looks like you are getting the variable extents here (_cacheParams.domainMin, _cacheParams.domainMax), and the only place you are using them is to pass them into SliceGridAlongPlane(), which also takes a grid3d argument. Why not simply call grid3d->GetUserExtents() from inside SliceGridAlongPlane()? The results should be identical, and it would simplify the code a good bit.

Reply:
This is because I need the full extents of a variable, not the extents of the current Grid, which may have had its region reduced.

The promise that the SliceGridAlongPlane makes is that it samples a variable along a plane throughout the entire domain.  Samples that are outside of the Box are counted as missingValues.  What I'm doing here might be a little unclear, so I'll write a note below discussing the strategy.

Reply:
Got it.